dwds/lib/src/debugging/metadata/module_metadata.dart

```
@@ -143,9 +146,13 @@ class ModuleMetadata {
         closureName = json['closureName'] as String,
         sourceMapUri = json['sourceMapUri'] as String,
         moduleUri = json['moduleUri'] as String {
-    var fileVersion = json['version'] as String;
-    if (!ModuleMetadataVersion.current.isCompatibleWith(version)) {
-      throw Exception('Unsupported metadata version $fileVersion');
+    if (!ModuleMetadataVersion.current.isCompatibleWith(version) &&
+        !ModuleMetadataVersion.previous.isCompatibleWith(version)) {
+      throw Exception('Unsupported metadata version $version. '
+          '\n  Supported versions: '
+          '\n    ${ModuleMetadataVersion.current.version} '
+          '\n    ${ModuleMetadataVersion.previous.version} '
+          '\n  Please upgrade dwds package.');
```

Generally people do not depend directly on `dwds` so this message likely will confuse them - I am not sure what the best thing would be to say since this could appear in the context of flutter, webdev, and internal...

---

Hmm you are right. This might happen if the SDK is updated but the clients have not. I'll remove the message and  will let the clients catch the exception and add an appropriate message instead.

---

Sounds good, we should put some sort of reasonable message in the `toString` of the exception as well for the cases where they don't handle it themselves.

---

Removed the "update dwds" part and removed catching the exception in MetadataProvider. If the exception happens, it is a bug so it is up to client to catch it (or just fail). 

The idea here is to prepare for the coming metadata update in SDK. This change needs to propagate to dwds,  webdev, and flutter before the SDK metadata version is updated. The original code ignored exceptions on metadata read so the debugger would start but fail to set breakpoints or do expression evaluation later. 

This change makes debugger fail to start if the metadata reading fails - does it sound reasonable?

---

I'll wait for @grouma review as well. Gary, it looks to me that the broken metadata you have encountered was just an empty string after the trailing line end, I fixed that. Did you see other malformed metadata?